cabal-install/Distribution/Client/Dependency.hs

```diff
@@ -452,35 +444,68 @@ removeLowerBounds (AllowOlder allowNewer)    params =
 
     relaxDeps :: UnresolvedSourcePackage -> UnresolvedSourcePackage
     relaxDeps srcPkg = srcPkg {
-      packageDescription = relaxPackageDeps removeLowerBound allowNewer
+      packageDescription = relaxPackageDeps relKind relDeps
                            (packageDescription srcPkg)
       }
 
 -- | Relax the dependencies of this package if needed.
 --
--- Helper function used by 'removeLowerBound' and 'removeUpperBounds'
-relaxPackageDeps :: (VersionRange -> VersionRange)
+-- Helper function used by 'removeBounds'
+relaxPackageDeps :: RelaxKind
                  -> RelaxDeps
                  -> PD.GenericPackageDescription -> PD.GenericPackageDescription
-relaxPackageDeps _ RelaxDepsNone gpd = gpd
-relaxPackageDeps vrtrans RelaxDepsAll  gpd = PD.transformAllBuildDepends relaxAll gpd
+relaxPackageDeps _ RelaxDepsNone gpd = gpd -- subsumed by no-op case in 'removeBounds'
+relaxPackageDeps relKind RelaxDepsAll  gpd = PD.transformAllBuildDepends relaxAll gpd
   where
-    relaxAll = \(Dependency pkgName verRange) ->
-      Dependency pkgName (vrtrans verRange)
-relaxPackageDeps vrtrans (RelaxDepsSome allowNewerDeps') gpd =
+    relaxAll (Dependency pkgName verRange) =
+        Dependency pkgName (removeBound relKind RelaxDepModNone verRange)
```

Comment:
Well, I do plan to support `--allow-newer=all`, `--allow-newer=^all`, `--allow-newer=foo:^all`, etc. via a future PR. That's a perfectly valid use-case  when I want to quickly see if there's overly constraining upper bounds in a specific package.

I'm not sure what to do about `zsh` getting in the way... which unquoted syntax does it _not_ try to interpret?  ;-)